Janus.js

```diff
@@ -0,0 +1,85 @@
+const fs = require("fs");
+const args = process.argv.slice(2);
+
+let t0 = Date.now();
+
+// Sample CLI input: node ./data/large1.txt 100
+
+fs.readFile(args[0], (err, data) => {
+	if (err) return console.log({ error: err });
+	else {
+		Janus(data, args[1]);
+		let t1 = Date.now();
+		let elapsed = t1 - t0;
+		console.log({ elapsed });
+		console.log("seconds elapsed = " + Math.floor(elapsed / 1000));
+	}
+});
+
+// 1st number = option // Aspect Name
+// 2nd number = cost // Negative Aspect
+// 3rd number = kudos // Positive Aspect
+
+// Janus will analyze data to maximize the total positive aspect
+// of positive aspects bound to a negative aspect(given a limit of potential negative aspect).
+
+const Janus = (data, maxCost) => {
+	data = data.toString();
+	const regex = /(\d{1,})/g;
+	data = data.match(regex);
+
+	let ranking = [];
+	for (let i = 2; i <= data.length - 1; i += 3) {
+		ranking.push({
+			option: data[i - 2],
+			cost: parseInt(data[i - 1]),
+			kudos: parseInt(data[i]),
+			weight: parseInt(data[i]) / parseInt(data[i - 1])
+		});
+	}
+
+	ranking = quickSort(ranking, "weight").reverse();
+
+	let bestOptions = [];
+	let currentCost = 0;
+	let currentKudos = 0;
+	let i = 0;
+	while (currentCost <= maxCost && i <= ranking.length - 1) {
+		currentCost += ranking[i].cost;
```

Comment:
instead of pushing in the current cost it should check if the next cost would make it over max (then just skip to the next iteration).

Reply:
yeah that was just a quick something to get it running because I wanted to see if it worked or not. I understand the rubber band effect my algorithm has with going over and the cutting the end off with those while loops. This creates that variance you see were my value figures are always a little under the true max. I have some thoughts on how to fix this but I like how fast it is at the moment :).